src/group_theory/quotient_group.lean

```diff
@@ -412,4 +412,57 @@ monoid_hom.to_mul_equiv
 
 end third_iso_thm
 
+
+section trivial
+
+/-These first few lemmas should maybe go somewhere else, where?-/
+
+variables {α : Type u} (r : α → α → Prop )
+
+lemma subsingleton_quot_iff_forall : subsingleton (quot r) ↔ ∀ a b, eqv_gen r a b :=
+begin
+  rw subsingleton_iff,
+  refine (surjective_quot_mk _).forall.trans (forall_congr $ λ a, _),
+  refine (surjective_quot_mk _).forall.trans (forall_congr $ λ b, _),
+  exact quot.eq,
+end
+
+lemma quot_by_top_is_subsingleton : subsingleton (quotient_group.quotient (⊤ : subgroup G)) :=
+trunc.subsingleton
+
+lemma left_rel_triv (H : subgroup G) (h : (quotient_group.left_rel H).r = λ _ _, true  ) :
+  ∀ x y : G , x⁻¹ * y ∈ H :=
+begin
+  let s := (quotient_group.left_rel H).r,
+  have h2 : ∀ x y : G, s x y ↔ x⁻¹ * y ∈ H, by {intros x y, refl},
+  intros x y,
+  have h3 := h2 x y,
+  rw ← h3,
+  simp_rw s,
+  rw h,
+  tauto,
+end
+
+/--If the quotient by a subgroup gives a singleton then the subgroup is the whole group-/
+
+lemma quot_subsingleton_triv  (H : subgroup G)
+ (h : subsingleton (quotient_group.quotient H)) : H = ⊤ :=
+begin
+  ext1,
+  simp only [subgroup.mem_top, iff_true] at *,
+  rw quotient_group.quotient at h,
+  let r := (quotient_group.left_rel H).r,
+  have := subsingleton_quot_iff_forall r,
```

Comment:
I think it would be easier to use `subsingleton_quotient_iff_eq_top` here

Reply:
I'm stuck with orange bars here so I can't fix this yet.

Reply:
Sure, you'd probably do best to wait for github to finish building then run `get-cache` rather than building locally.

Reply:
It seems I got lucky and the bars went away, its working now.